spdata/reader.py

```diff
@@ -45,3 +45,26 @@ def load_txt(content) -> ty.Dataset:
     coordinates = ty.Coordinates(*zip(*coordinates))
 
     return ty.Dataset(data, coordinates, mzs, labels)
+
+def load_imzml(file_path) -> ty.Dataset:
+    """Load Dataset from imzml file
+
+    Args:
+        file_path: path to imzml file
+
+    Returns:
+        out: spdata.types.Dataset
+    """
+    with imzparse(file_path) as input_handle:
+        assert np.min(input_handle.mzLengths) == np.max(input_handle.mzLengths)
```

Comment:
If we want to go temporarily for such a check, it would probably be better to have this enabled also when optimizations are turned on. It is now the only heuristics that allows us to read only one type of imzML files, but this should be known to the user, not pass silently when optimized out. I'd propose something like:
```
if np.min(input_handle.mzLengths) != np.max(input_handle.mzLengths):
    raise ValueError("Can't read processed / continuous.") # I do not remember which was which.
```

Reply:
👍